### PR TITLE
Connect Nft Picker to Existing safes

### LINKED ISSUE #3835: Connect Transfer NFT to Existing Safes
This issue is a sub-issue of #3722.

## This issue is complete when...
 - [ ] Selecting a safe populates the NFT selector with the NFT stored in the safe.
 - [ ] The NFT displays the NFT image, contract, and id.

### PR DESCRIPTION
**Pull request by willm30**
## Description

This PR connects the Nft Picker to the existing Safes in the colony. 

You can add the following safes with NFTS to your colony in order to test (both Rinkeby):

1. [0x05b990D01A2f86852feD1E7d609dd3E52aA9EA27](https://gnosis-safe.io/app/rin:0x05b990D01A2f86852feD1E7d609dd3E52aA9EA27)
1. [0x358E6D7C8Fa5e55fc6A0D50E23b875c6c49fF09A](https://gnosis-safe.io/app/rin:0x358E6D7C8Fa5e55fc6A0D50E23b875c6c49fF09A)

And this one doesn't have NFTs (also Rinkeby):
1. 0x0DA0C3e52C977Ed3cBc641fF02DD271c3ED55aFe

To test:
1. Add a safe with and without nfts
3. Go to safe control
4. Select a safe and select "Transfer NFT"
5. Should display loading state while we fetch nft data. If there's none, should display relevant message. If there's an error fetching (e.g. no internet connection), should also display relevant message. 
6. Once data is fetched, you should be able to select an nft in the safe. If it has an image uri, this should show in the avatar component, and the contract name and id should be correctly inputted. 

It also does a little bit of housekeeping:
1. Removing "testnet" from Goerli and Rinkeby. I added these in an earlier PR thinking they'd be useful but they just take up unnecessary space in the safe picker input and it's also not how Gnosis Safe refers to them. So reverting.
7. Updates validation schema for `safe`
8. renamed `getSafeCoreSDK.ts` to `safeHelpers.ts` and added a few utility functions.

**Changes** 🏗

* Wires up the NFT picker to the safes that exist in the colony.

## Screenshot

nfta

Resolves #3835
Contributes to #3722

### CODE REVIEW

**Comment by willm30** [comment]
> Not a full-scale review but I noticed you use both `Nft` and `NFT` which seems a bit inconsistent

@jakubcolony Can you leave an in-line comment in the code where you see this?

File: src/modules/dashboard/sagas/utils/safeHelpers.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  );
+
+  if (!selectedNetwork || !selectedNetwork.safeTxService) {
+    throw new Error(`Selected chain ${selectedChain} not currently supported.`);
+  }
+
+  return selectedNetwork.safeTxService;
+};
+
+export const getIdFromNFTDisplayName = (displayName: string) => {
```
Comment (jakubcolony) [inline src/modules/dashboard/sagas/utils/safeHelpers.ts:60]: `NFT`

File: src/modules/dashboard/components/Dialogs/ControlSafeDialog/TransactionTypesSection/TransferNFTSection.tsx [hunk 2]
```diff
@@ -66,27 +85,89 @@ const renderAvatar = (address: string, item) => (
...
+
+  const [
+    { value: selectedNft },
+    ,
+    { setValue: setSelectedNft },
+  ] = useField<SelectedNft | null>(`transactions.${transactionFormIndex}.nft`);
+
+  useEffect(() => {
+    setSelectedNft(null);
+    const getNFTs = async (chosenSafe: SelectedNft) => {
```
Comment (jakubcolony) [inline src/modules/dashboard/components/Dialogs/ControlSafeDialog/TransactionTypesSection/TransferNFTSection.tsx:123]: `NFT`

**Review by jakubcolony — COMMENTED** [review]
The existing code seems to favour `NFT` while most of the stuff you added is `Nft`. I highlighted two files in which you use `NFT`. Probably not a big deal but hope this helps

**Review by Julianahlk — COMMENTED** [review]
Nice work @willm30! 

Screen Shot 2022-09-13 at 1 09 04 PM
Screen Shot 2022-09-13 at 1 09 21 PM
Screen Shot 2022-09-13 at 1 09 33 PM
Screen Shot 2022-09-13 at 1 20 58 PM
Screen Shot 2022-09-13 at 1 15 46 PM

Question regarding the following:

1. I added this safe [0x358E6D7C8Fa5e55fc6A0D50E23b875c6c49fF09A](https://gnosis-safe.io/app/rin:0x358E6D7C8Fa5e55fc6A0D50E23b875c6c49fF09A) but I didn't see any NFT image loaded, does this NFT have an image?

Screen Shot 2022-09-13 at 1 21 10 PM

creature

2. I was testing what happens if you confirm a transaction and hit the back button to go back to the control safe dialog and when I go back I see the state of the dialog is a bit different. I do not have any NFT selected, image and the transaction type is gone, is this normal?
NFTTESTING

**Review by ArmandoGraterol — CHANGES_REQUESTED** [review]
> I think that's actually the NFT lol
FireShot Capture 719 - Safe - gnosis-safe io

I see, I was expecting an actual NFT creature 😂

**Comment by Julianahlk** [comment]
> Excellent work Will! Left some comments but overall good job wiring this up
> 
> @Julianahlk
> 
> > I added this safe [0x358E6D7C8Fa5e55fc6A0D50E23b875c6c49fF09A](https://gnosis-safe.io/app/rin:0x358E6D7C8Fa5e55fc6A0D50E23b875c6c49fF09A) but I didn't see any NFT image loaded, does this NFT have an image?
> 
> I think that's actually the NFT lol
> 
> > I was testing what happens if you confirm a transaction and hit the back button to go back to the control safe dialog and when I go back I see the state of the dialog is a bit different. I do not have any NFT selected, image and the transaction type is gone, is this normal?
> 
> Can confirm it's happening, and it's not normal behaviour

Sorry, I think I commented and edited the message above by accident. But thanks for clarifying. I was expecting the image of a creature.

**Comment by willm30** [comment]
@ArmandoGraterol @Julianahlk Should now be fixed.

File: src/modules/dashboard/components/Dialogs/ControlSafeDialog/ControlSafeForm.tsx [hunk 3]
```diff
@@ -35,6 +35,7 @@ import {
 import { TransactionTypes, transactionOptions } from './constants';
 
 import styles from './ControlSafeForm.css';
+import { SelectedSafe } from '~modules/dashboard/sagas/utils/safeHelpers';
```
Comment (ArmandoGraterol) [inline src/modules/dashboard/components/Dialogs/ControlSafeDialog/ControlSafeForm.tsx:38]: Above or else :gun::sunglasses:

**Review by ArmandoGraterol — COMMENTED** [review]
Right now if you change from 1 safe with NFT to another with NFT there's like 1-2 seconds of delay until the NFTs are updated, meanwhile, the NFTs of the previous safe are still showing in the NFT picker dropdown. This delay may be longer in QA/prod.

We should check that the NFT picker options are also reinitialized when the value changes

**Comment by willm30** [comment]
@ArmandoGraterol what's the problem? The NFT picker is reinitialised when the safe value changes.

nfts

**Review by Julianahlk — APPROVED** [review]
Nice job! NFTs are now updating according to which safe is selected.

**Comment by ArmandoGraterol** [comment]
> @ArmandoGraterol what's the problem? The NFT picker is reinitialised when the safe value changes.

Sorry, the second paraph must've muddied what I tried to say, I'm talking about this:

[823b57f1-1646-4339-9ec8-121096e9337e.webm](https://user-images.githubusercontent.com/18473896/190522297-a80623d4-a7cf-4d51-902f-f281cfbfd39a.webm)

I can see this delay sometimes in my dev env without any throttling. I'm expecting that in QA/prod this delay will be longer, that's why I think that if you change safes then we should also display spinner until the data is updated, just in case

**Comment by willm30** [comment]
@ArmandoGraterol I see, thanks. I've added the spinner and a cache so we don't make calls after the first time.

**Review by ArmandoGraterol — APPROVED** [review]
Looking great to me :ok_hand: Just have a question regarding the `setIsLoadingNFTData` but after that's addressed this should be good to go

File: src/modules/dashboard/components/Dialogs/ControlSafeDialog/TransactionTypesSection/TransferNFTSection.tsx [hunk 4]
```diff
@@ -66,27 +87,134 @@ const renderAvatar = (address: string, item) => (
...
+          setIsLoadingNFTData(() => true);
+          const nftData = await getNFTs(safe);
+          setAvailableNFTs(nftData);
+          if (nftData) {
+            setSavedNFTs({
+              ...savedNFTs,
+              [safe.id]: nftData,
+            });
+          }
+          setIsLoadingNFTData(() => false);
```
[the remark this answers is not in the document]
Comment (willm30) [reply src/modules/dashboard/components/Dialogs/ControlSafeDialog/TransactionTypesSection/TransferNFTSection.tsx]: @ArmandoGraterol Yes you're right, have changed.

**Pull request merged**